aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -2709,6 +3271,542 @@ module aptos_framework::delegation_pool {
+        assert!(calculate_and_update_voter_total_voting_power(pool_address, delegator2_address) == 90 * ONE_APT, 1);
+
+        // Delegator1 delegates its voting power to voter1. Delegator2 delegates its voting power to voter2.
+        // It takes 1 cycle to take effect. No immediate change.
+        delegate_voting_power(delegator1, pool_address, voter1_address);
+        delegate_voting_power(delegator2, pool_address, voter2_address);
+        assert!(calculate_and_update_voter_total_voting_power(pool_address, voter1_address) == 0, 1);
+        assert!(calculate_and_update_voter_total_voting_power(pool_address, voter2_address) == 0, 1);
+        assert!(calculate_and_update_voter_total_voting_power(pool_address, delegator1_address) == 10 * ONE_APT, 1);
+        assert!(calculate_and_update_voter_total_voting_power(pool_address, delegator2_address) == 90 * ONE_APT, 1);
+
+        // One cycle passed. The voter change takes effects.
+        timestamp::fast_forward_seconds(LOCKUP_CYCLE_SECONDS);
+        end_aptos_epoch();
+        assert!(calculate_and_update_voter_total_voting_power(pool_address, voter1_address) == 10 * ONE_APT, 1);
+        assert!(calculate_and_update_voter_total_voting_power(pool_address, voter2_address) == 90 * ONE_APT, 1);
+        assert!(calculate_and_update_voter_total_voting_power(pool_address, delegator1_address) == 0, 1);
+        assert!(calculate_and_update_voter_total_voting_power(pool_address, delegator2_address) == 0, 1);
+    }
+
+    #[test(aptos_framework = @aptos_framework, validator = @0x123, delegator1 = @0x010, delegator2 = @0x020, voter1 = @0x030, voter2 = @0x040)]
+    public entry fun test_voting_power_change_for_rewards(
+        aptos_framework: &signer,
+        validator: &signer,
+        delegator1: &signer,
+        delegator2: &signer,
+        voter1: &signer,
+        voter2: &signer,
+    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords {
+        initialize_for_test_custom(
+            aptos_framework,
+            100 * ONE_APT,
+            10000 * ONE_APT,
+            LOCKUP_CYCLE_SECONDS,
+            true,
+            100,
+            100,
+            1000000
+        );
+        aptos_governance::initialize_for_test(
+            aptos_framework,
+            (10 * ONE_APT as u128),
+            100 * ONE_APT,
+            1000,
+        );
+        aptos_governance::initialize_partial_voting(aptos_framework);
+        features::change_feature_flags(aptos_framework, vector[features::get_partial_governance_voting()], vector[]);
+
+        // 50% commission rate
+        initialize_test_validator_custom(validator, 100 * ONE_APT, true, false, 5000);
+
+        let validator_address = signer::address_of(validator);
+        let pool_address = get_owned_pool_address(validator_address);
+        // Delegation pool is created after partial governance voting feature flag is enabled. So this delegation
```

Comment:
Same here. Do we actually need two delegators?

Reply:
Actually here I want to test the case that multiple delegators delegate to the same voter and both of them receive rewards. Added the case.